core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/LogicalPlan.scala

```diff
@@ -23,7 +24,12 @@ object LogicalPlan {
       links: List[LogicalLink]
   ): DirectedAcyclicGraph[OperatorIdentity, LogicalLink] = {
     val workflowDag =
-      new DirectedAcyclicGraph[OperatorIdentity, LogicalLink](classOf[LogicalLink])
+      new DirectedAcyclicGraph[OperatorIdentity, LogicalLink](
+        null,
```

Comment:
Please add the key names to those parameters.

Reply:
I tried but we cannot add key names here. This constructor is a java function, and for this library it cannot be called with named parameters. (See [this post](https://stackoverflow.com/questions/53765853/scala-named-method-arguments-cannot-resolve-symbol).) I'll add a comment here instead.